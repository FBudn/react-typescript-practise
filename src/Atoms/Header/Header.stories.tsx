import React from "react";
import { ComponentMeta, Story } from "@storybook/react";
import Header, { HeaderProps } from "./Header";

export default {
  title: "Atoms/Header",
  component: Header,
} as ComponentMeta<typeof Header>;

const Template: Story<HeaderProps> = (args) => <Header {...args} />;
export const FlexFullWidthLOGIN = Template.bind({});

FlexFullWidthLOGIN.args = {
  children: "LOGIN",
};

export const FlexFullWidthSIGNUP = Template.bind({});

FlexFullWidthSIGNUP.args = {
  children: "SIGN UP",
};
