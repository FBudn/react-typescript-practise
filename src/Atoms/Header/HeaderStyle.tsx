/* eslint-disable import/prefer-default-export */
import styled from "styled-components";

export const HeaderContainer = styled.div`
  display: flex;
  width: 100%;
  justify-content: start;
  align-items: center;
  font-weight: bold;
  color: rgb(82 82 91);
`;
