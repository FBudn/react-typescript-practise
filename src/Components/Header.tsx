import React from "react";
import styled from "styled-components";

const HeaderContainer = styled.div`
  display: flex;
  width: 100%;
  justify-content: start;
  align-items: center;
  font-weight: bold;
  color: rgb(82 82 91);
`;

export interface HeaderProps {
  text: string;
}

const Header: React.FC<HeaderProps> = ({ text }) => {
  return <HeaderContainer>{text}</HeaderContainer>;
};

export default Header;
