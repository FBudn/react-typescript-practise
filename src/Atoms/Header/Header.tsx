import React from "react";
import { HeaderContainer } from "./Header.style";

export interface HeaderProps {
  children: string;
  testId?: string;
}

const Header: React.FC<HeaderProps> = ({ children, testId }) => {
  return <HeaderContainer data-testid={testId}>{children}</HeaderContainer>;
};

export default Header;
