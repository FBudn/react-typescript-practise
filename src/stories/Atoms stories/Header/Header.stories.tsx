import React from "react";
import { ComponentMeta, Story } from "@storybook/react";
import Header, { HeaderProps } from "../../../Atoms/Header/Header";

export default {
  title: "Atoms/Header",
  component: Header,
} as ComponentMeta<typeof Header>;

const Template: Story<HeaderProps> = (args) => <Header {...args} />;
export const FlexFullWidth = Template.bind({});

FlexFullWidth.args = {
  children: "LABEL: Label which as a default has global text styles",
};
