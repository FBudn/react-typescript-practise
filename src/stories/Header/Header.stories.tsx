import React from "react";
import { Meta, Story } from "@storybook/react";
import Header, { HeaderProps } from "../../Atoms/Header/Header";

export default {
  title: "Atoms/Header",
  component: Header,
} as Meta;

const Template: Story<HeaderProps> = (args) => <Header>{args.children}</Header>;
export const FlexFullWidth = Template.bind({});
FlexFullWidth.args = {
  children: "LABEL: Label which as a default has global text styles",
};
