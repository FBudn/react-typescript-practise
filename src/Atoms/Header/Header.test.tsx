import React from "react";
import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Header, { HeaderProps } from "./Header";
import "jest-styled-components";

const props: HeaderProps = {
  children: `Header Test`,
  testId: `test-header-id`,
};

test("Should render Header component", () => {
  render(<Header {...props} />);
  const HeaderElement = screen.getByTestId(`test-header-id`);
  expect(HeaderElement).toBeInTheDocument();
});

test("Should match snapshot:", () => {
  const HeaderElement = TestRenderer.create(<Header {...props} />).toJSON();
  expect(HeaderElement).toMatchSnapshot();
});

test("Should render and have styles:", () => {
  render(<Header {...props} />);
  const HeaderElement = screen.getByTestId(`test-header-id`);
  expect(HeaderElement).toBeInTheDocument();

  expect(HeaderElement).toHaveStyleRule("display", "flex");
  expect(HeaderElement).toHaveStyleRule("width", "100%");
  expect(HeaderElement).toHaveStyleRule("justify-content", "start");
  expect(HeaderElement).toHaveStyleRule("align-items", "center");
  expect(HeaderElement).toHaveStyleRule("display", "flex");
});

test(`Should render with text`, () => {
  render(<Header {...props} />);
  const HeaderElement = screen.getByTestId(`test-header-id`);
  expect(HeaderElement).toBeInTheDocument();
  expect(HeaderElement).toHaveTextContent(`Header Test`);
});
