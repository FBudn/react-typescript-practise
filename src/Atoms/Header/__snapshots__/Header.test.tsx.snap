// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot and have styles: 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  color: rgb(82 82 91);
}

<div
  className="c0"
  data-testid="Header-1"
>
  Header Test
</div>
`;
